h/api/nipsa/search.py

```diff
@@ -0,0 +1,83 @@
+import copy
+
+
+def nipsa_filter(query, userid=None):
```

Comment:
Nonsense. The only places that call this are `streamer` and `search`. The `streamer` code already knows _far_ _far_ too much about Elastisearch. That would only ever change, maybe, if we had it use some totally abstracted search call, which maybe `api` already exposes well enough. And `search` is already constructed the query object anyway in order to pass that in.

Basically, if the caller has to know about the structure of an ES query in order to pass it to this function I find the claim that having the caller instead receive the filter and have to construct an ES query from it is any more exposing details about ES totally bogus.